src/ret_val.h

```diff
@@ -0,0 +1,64 @@
+#pragma once
+#ifndef RET_VAL_H
+#define RET_VAL_H
+
+#include "debug.h"
+
+#include <string>
+#include <type_traits>
+
+template<typename T, typename std::enable_if<std::is_same<T, bool>::value, int>::type = 0>
+struct def_success_const : std::integral_constant<bool, true> {};
```

Comment:
At least one comment for those cases would be useful.
Not everyone here can understand those big C++ template definition lines at a glance.
Something simple, like "For a given class, this defines value that should be treated as `true` for purposes of `ret_val` cast to `bool`" (if that is what it does).

Reply:
Yes, that was their purpose, but they were over-engineered. See [300faf4](https://github.com/CleverRaven/Cataclysm-DDA/pull/21774/commits/300faf4895319ba500e270436cc4ac6971fd69d8#diff-292991262ce305fd8d128477e02f4e91R67). The template probably doesn't need a comment anymore.